wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet/Delete.hs

```diff
@@ -15,10 +17,22 @@ import           Cardano.Wallet.Kernel.DB.Util.AcidState
   DELETE
 -------------------------------------------------------------------------------}
 
+-- | Errors thrown by 'deleteHdAccount'
+data DeleteHdAccountError =
+    -- | Parent root not found
+    DeleteHdAccountUnknownRoot UnknownHdRoot
+  deriving Eq
+
```

Comment:
I don't think we need custom error types for each operation. I'd only introduce them where needed (see also comments on `UpdateHdAccountError`, below).

Reply:
I almost felt this was necessary to play nicely with the errors we report upstream, but if you are strongly against it, I can scrap it.

Reply:
I started down this path initially when setting up this infrastructure, and ended up with waaaay too many types -- and then realized they were mostly just isomorphic copies. Pointless exercise.